### DISCUSSION TITLE
make form for array casted model.

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#7055] [Help] [answered]
I have Setting model which has `$casts`. 

```php

class Setting extends Model
{
    protected $fillable = [
        'content',
    ];

    protected $casts = [
        'content' => 'array',
    ];
}

```
I would like to make form to save this kinds of array data into 'content' field.

```php 
[
    'points'=>[
        'write'=>10,
        'comment'=>20,
    ],
    'access'=>[
        'index'=>true,
        'write'=>false,
    ]
]
```

How can I do make form for this array data?

**Comment by pboivin** [comment, marked as answer]
I think something like this should work:
```php
Group::make([
    TextInput::make('points.write'),
    TextInput::make('points.comment'),
    TextInput::make('access.index'),
    TextInput::make('access.write'),
])->statePath('content'),
```

**Reply by magic-thomas** [reply]
Thank you. it works. 

But I think I should use `Toggle` component instead of `TextInput`  for boolean type data. And I found `Toggle` component does not support deep array accessing. 

You can check https://github.com/filamentphp/filament/issues/7056. 

Now I am using `Radio` Component which allows deep array accessing.

Thanks.

**Reply by danharrin** [reply]
Toggle should support it

**Reply by danharrin** [reply]
Open an issue with a reproduction repo and we will look at it
